src/main/java/Parser.java

```diff
@@ -0,0 +1,38 @@
+import java.util.Scanner;
+
+/**
+ * Represents the parsing of user commands into actual command objects.
+ * 		Basically a switch for deciding which commands to execute based on the input
+ */
+public class Parser {
+	public Parser() { }
+
+	/**
+	 * parse takes in the user's input as a fullCommand and decideds which command object to initialize based on the
+	 * 		first word of the fullCommand
+	 * @param fullCommand the entire command the user has input into the duke program
+	 * @return returns the corresponding command with the user's input
+	 * @throws DukeException throws exception if the command is not found.
+	 */
+	public static Command parse(String fullCommand) throws DukeException {
```

Comment:
Your current implementation of the `Parser` class only does not actually do much parsing of the command string apart from its type. Consider refactoring your `Parser` class to handle most (if not all) of the parsing of the command string, including the individual arguments in a command. This will allow you to extract a lot of the business logic from your `Command` sub-classes.